app/src/main/java/edu/wisc/ece/pinpoint/pages/profile/ProfilePageFragment.java

```diff
@@ -134,11 +134,18 @@ public void onPageSelected(int position) {
         });
     }
 
-    public void setUserData(@NonNull User user) {
+    public void setUserData(@NonNull User user, String uid) {
         user.loadProfilePic(profilePic, this);
         username.setText(user.getUsername());
-        followerCount.setText(String.valueOf(user.getNumFollowers()));
-        followingCount.setText(String.valueOf(user.getNumFollowing()));
+        if (uid.equals(firebase.getCurrentUser().getUid())) {
+            // if the profile belongs to the app user, display the cached values
+            followerCount.setText(String.valueOf(firebase.getCachedFollowerIds().size()));
+            followingCount.setText(String.valueOf(firebase.getCachedFollowingIds().size()));
+        }
+        else {
+            followerCount.setText(String.valueOf(user.getNumFollowers()));
+            followingCount.setText(String.valueOf(user.getNumFollowing()));
+        }
```

Comment:
Let's use the `numFollowers` and `numFollowing` fields for both the current user and other users rather than doing this conditional check since those stats should always be up to date.